### DISCUSSION TITLE
PTZ Pelco-P

### DISCUSSION THREAD

**Discussion by bjct42** [ZoneMinder/zoneminder#3979] [General]
Hello, my camera is equipped with the PelcOP PTZ protocol. After configuring the camera control option the PTZ does not work:
/usr/bin/zmcontrol.pl --id 1 
30/04/2024 07:50:41.617680 zmcontrol_1[292380].INF [main:116] [Starting control server 1/PelcoP]
30/04/2024 07:50:41.618583 zmcontrol_1[292380].INF [main:127] [Control server 1/PelcoP starting at 24/04/30 07:50:41]
Can't call method "baudrate" on an undefined value at /usr/share/perl5/vendor_perl/ZoneMinder/Control/PelcoP.pm line 58.
Can you help me?

**Comment by connortechnology** [comment]
It is failing to open the serial port.  Did you tell it which one to use in ControlDevice field?  Could also be a permissions problem.

**Comment by bjct42** [comment]
I did not understand what to put in Control Device, the name of the camera, its ID or other ? currently I put the name of the camera.

**Comment by bjct42** [comment]
Precision, my carmera is an IP camera with integrated PTZ

**Comment by connortechnology** [comment]
This PelcoP stuff is really old.  It is trying to talk to the camera of a serial link, not network.  A valid value would have been something like /dev/ttyS1.  

What is the model of camera?

**Comment by bjct42** [comment]
China Cam (xiongmaitech model :X6-WEQ). Polyvision software tells me that the PTZ protocol is Pelcod. But I don’t have any more specific information. Is there a scanner that would allow to know the PTZ protocol used?

**Comment by bjct42** [comment]
I just solved my problem, I had to use the ONVIF protocol. Thank you for the help

**Comment by bjct42** [comment]
But I cannot edit the PTZ protocol, when I click on EDIT, I have a blank page without error

**Comment by connortechnology** [comment]
I fixed an issue like that this morning. Try an update.

**Comment by bjct42** [comment]
after the update to 1.37.59-1
Options/Control and Edit Control =>  OK
From camera, Edit/Control and Edit Control Type KO => (blank page)

**Comment by bjct42** [comment]
However, after modifying a Crontol (ONVIF Cam for example) or creating a new Control identical to ONVIF Cam, but with another name, the PTZ no longer works
Can't access name:moveRelRight AUTOLOAD:ZoneMinder::Control::onvif::moveRelRight member of object of class ZoneMinder::Control::onvif from main:171
